sdk/trace/provider.go

```diff
@@ -274,31 +281,52 @@ func WithSpanProcessor(sp SpanProcessor) TracerProviderOption {
 	}
 }
 
-// WithResource option attaches a resource to the provider.
-// The resource is added to the span when it is started.
+// WithResource returns a TracerProviderOption that will configure the
+// Resource r as a TracerProvider's Resource. The configured Resource is
+// referenced by all the Tracers the TracerProvider creates. It represents the
+// tracing environment the TracerProvider operates in.
```

Comment:
The term "tracing environment" is evocative. Does that come from the specification? I recall hearing that a resource is meant to represent something about the publishing entity (where it's running, what it's representing, and so on). That could be the "environment," but I wonder whether there are other terms we could borrow from the specification.

I apologize if you already have this right. I've had trouble understanding what "resources" are since first coming to OpenTelemetry.

Reply:
What about ...

```suggestion
// referenced by all the Tracers the TracerProvider creates. It represents the
// entity producing telemetry that the TracerProvider represents.
```

?

From the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.1.0/specification/resource/sdk.md#resource-sdk):

> A Resource is an immutable representation of the entity producing telemetry as Attributes.

Reply:
Yes, that's a nice adaptation.

Reply:
Or just remove the sentence?

Reply:
I'd keep it there. I prefer all the help I can get understanding and being reminded of what a resource is in this context.